File: week-1/problems/mario.c
#include <cs50.h>
#include <stdio.h>

int getHeight()
{
  return get_int("Enter a height between 1 and 8: \n");
}

int main(void)
{
  int height = -1;
  //loop until valid height
  while (height < 1 || height > 8)
  {
    height = getHeight();
  }

  //loop through every row
  for (int i = 1; i <= height; i++)
  {
    //left side of pyramid
    for (int j = 1; j <= height; j++)
    {
      char poundOrSpace = (j > height - i) ? '#' : ' ';
      printf("%c", poundOrSpace);
    }

    //gap
    printf("  ");

    //right side of pyramid
    for (int j = 1; j <= height; j++)
    {
      if (j <= i)
        printf("#");
    }

    printf("\n");
  }
}

File: week-6/problems/mario/mario.py
from cs50 import get_int

rows = -1

while (rows > 8 or rows < 1):
    rows = get_int('Please enter a number of rows > 0 and < 8. ')

for row in range(rows):
    # print space
    print(f" "*(rows-row-1), end = '')
    print(f"#"*(row+1))